Tasks/SonarQubePostTest/PRCA/PostComments-Module.psm1

```diff
@@ -41,56 +72,143 @@ function InitPostCommentsModule
 #
 # Initializes the module. 
 #
-# Remark: for test purposes only
+# Remark: for test only
 #
 function Test-InitPostCommentsModule
 {
-    param ([Microsoft.TeamFoundation.SourceControl.WebApi.GitHttpClient]$gitClient, 
+    param ([Microsoft.TeamFoundation.SourceControl.WebApi.GitHttpClient]$gitClient,
+    [Microsoft.VisualStudio.Services.CodeReview.Discussion.WebApi.DiscussionHttpClient]$discussionClient, 
     [Microsoft.TeamFoundation.SourceControl.WebApi.GitPullRequest]$pullRequest, 
     [string]$artifactUri)
     
     $script:gitClient = $gitClient
+    $script:discussionClient = $discussionClient
     $script:pullRequest = $pullRequest
     $script:artifactUri = $artifacturi
 }
 
 #
-# Posts new comments, ignoring duplicate comments and resolves comments that were open in an old iteration of the PR
+# Posts new messages, ignoring duplicate comments and resolves comments that were open in an old iteration of the PR.
+# Comment source is used to decorate comments created by this logic. Only comments with the same source will be resolved.
 #
 function PostAndResolveComments
 {
-    param ([Array][ValidateNotNull()]$comments)
+    param ([Array][ValidateNotNull()]$messages, [string][ValidateNotNullOrEmpty()]$commentSource)
     
-    ValidateComments $comments
+    ValidateMessages $messages
     
-    Write-Host "Processing $($comments.Count) new comments"
+    Write-Host "Processing $($messages.Count) new messages"
     
     #TODO: ResolveExistingIssues
     
-    if ($comments.Count -gt 0)
+    if ($messages.Count -gt 0)
     {
-        InternalPostNewComments $comments
+        InternalPostNewMessages $messages $commentSource
     }
 }
 
+#endregion
+
+#region Private
+
+
 #
 # Posts the discussion threads loaded with comments to the PR
-# Remark: public for test purposes
+#
 function PostDiscussionThreads
```

Comment:
I decided to properly mock these to improve unit test coverage and to ease testing when it comes to posting multiple iterations of messages